### ISSUE TITLE
Dereferencing a null struct throws errors.

### ISSUE THREAD

**Issue by jnh5y** [confluentinc/ksql#9029]
**Describe the bug**
If a struct is `null`, dereferencing it causes `NullPointerException` and logs errors.

**Expected behavior**
We could add a null check before dereferencing a struct.

**Comment by Eallison123** [comment]
Hi, I just attempted to reproduce this bug and was unable. I would like to help fix this bug so here are the steps I tried to reproduce it.
```
CREATE STREAM NULL_POINTER(
id VARCHAR,
obj STRUCT<value_to_be_null VARCHAR,
value_to_be_filled_in VARCHAR>)
WITH (KAFKA_TOPIC='topic_name',
VALUE_FORMAT='JSON',
PARTITIONS=1);
```
I then inserted with:
```
INSERT INTO NULL_POINTER (id, obj) VALUES ('some value', null);
```
and queried it with:
```
SELECT id, obj->VALUE_TO_BE_NULL FROM NULL_POINTER;
```
any my results were:
+----------------------------------------------------------+----------------------------------------------------------+
|ID                                                        |VALUE_TO_BE_NULL                                          |
+----------------------------------------------------------+----------------------------------------------------------+
|some value                                                |null                                                      |
Am I recreating this incorrectly or was the bug fixed?
Thanks

**Comment by jnh5y** [comment]
Hi @Eallison123, at this point, I believe the issue that remains is that there's a message in the logs about the failure.

If you'd like to work on it, I'd be happy to point you in the right direction!  (I was thinking about picking up these issues since we are thinking about cutting the next release soonish and this issue should be a quick one.)

**Comment by Eallison123** [comment]
@jnh5y that would be great, I didn't realize there were logs for the client (although I should have assumed). Any help would be appreciated, if you'd rather dm or just post on this thread that would be amazing.

**Comment by jnh5y** [comment]
@Eallison123 are you on Gitter.im or the Confluent Community Slack?  Either of those would be places we could chat through this more.  If I get a few minutes today, I may be able to toss up a quick PR for this.  If that happens, I'll post here!

**Comment by miguelskeeled** [comment]
Hi @Eallison123, I'm a KSQL user which has been having this issue.
To replicate the issue, your example should have a nested struct, like this:
```
CREATE STREAM NULL_POINTER(
id VARCHAR,
obj STRUCT<
   value_to_be_null STRUCT <
       value_to_fill VARCHAR
   >,
   value_to_be_filled_in VARCHAR>)
WITH (KAFKA_TOPIC='topic_name',
VALUE_FORMAT='JSON',
PARTITIONS=1);
```

Then, this query should throw an error log

```
SELECT id, obj->VALUE_TO_BE_NULL->VALUE_TO_FILL FROM NULL_POINTER;
```

**Comment by pgaref** [comment]
This should be fixed by #9127
